pull/github_context.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2018 Palantir Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pull
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+)
+
+// GithubContext is a Context implementation that gets information from GitHub.
+// A new instance must be created for each request.
+type GithubContext struct {
+	client *github.Client
+
+	owner  string
+	repo   string
+	number int
+	pr     *github.PullRequest
+
+	// cached fields
+	comments         []string
+	requiredStatuses []string
+	successStatuses  []string
+}
+
+func NewGithubContext(client *github.Client, pr *github.PullRequest, owner, repo string, number int) Context {
+	return &GithubContext{
+		client: client,
+
+		pr:     pr,
+		owner:  owner,
+		repo:   repo,
+		number: number,
+	}
+}
+
+func (ghc *GithubContext) Owner() string {
+	return ghc.owner
+}
+
+func (ghc *GithubContext) Repo() string {
+	return ghc.repo
+}
+
+func (ghc *GithubContext) Number() int {
+	return ghc.number
+}
+
+func (ghc *GithubContext) Locator() string {
+	return fmt.Sprintf("%s/%s#%d", ghc.owner, ghc.repo, ghc.number)
+}
+
+func (ghc *GithubContext) Title(ctx context.Context) (string, error) {
+	return ghc.pr.GetTitle(), nil
+}
+
+func (ghc *GithubContext) Body(ctx context.Context) (string, error) {
+	return ghc.pr.GetBody(), nil
+}
+
+func (ghc *GithubContext) Comments(ctx context.Context) ([]string, error) {
+	if ghc.comments == nil {
+
+		prCommentOpts := &github.PullRequestListCommentsOptions{ListOptions: github.ListOptions{PerPage: 100}}
```

Comment:
What was the reasoning behind supporting pull request review comments as part of the whitelist/blacklist checking? Is this only top-level review comments or does this include line comments as well?